### PR TITLE
MAISTRA-819 Fix SMCP status conditions

### PR DESCRIPTION
**Pull request by luksa**
To see how the events & conditions appear after applying these changes, see comment https://issues.jboss.org/browse/MAISTRA-819?focusedCommentId=13774056&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13774056

### CODE REVIEW

File: pkg/controller/servicemesh/controlplane/reconciler.go [hunk 1]
```diff
@@ -465,8 +481,10 @@ func (r *ControlPlaneReconciler) postReconciliationStatus(reconciliationMessage
 	}
 	reconciledCondition := r.Status.GetCondition(v1.ConditionTypeReconciled)
 	if processingErr == nil {
+		reconciledCondition.Reason = reconciliationReason
```
Comment (brian-avery) [inline pkg/controller/servicemesh/controlplane/reconciler.go]: Since this is true regardless of which condition we hit, should it be external to the if?

**Review by brian-avery — APPROVED** [review]

File: pkg/controller/servicemesh/controlplane/reconciler.go (hunk 1 above)
Comment (luksa) [reply pkg/controller/servicemesh/controlplane/reconciler.go]: Of course. Thanks! Fixed.

File: pkg/controller/servicemesh/controlplane/controller.go [hunk 2]
```diff
@@ -185,9 +185,36 @@ func (r *ReconcileControlPlane) Reconcile(request reconcile.Request) (reconcile.
...
+
 		reqLogger.Info("Deleting ServiceMeshControlPlane")
 		err := reconciler.Delete()
 		if err != nil {
+			reconciler.Status.SetCondition(v1.Condition{
+				Type:    v1.ConditionTypeReconciled,
+				Status:  v1.ConditionStatusFalse,
+				Reason:  v1.ConditionReasonDeletionError,
+				Message: fmt.Sprintf("Error deleting service mesh: %s", err),
+			})
```
Comment (dgn) [inline pkg/controller/servicemesh/controlplane/controller.go:218]: Isn't this missing a `reconciler.PostStatus()`?

Comment (luksa) [reply pkg/controller/servicemesh/controlplane/controller.go:218]: Great catch! I've fixed it now. And tested it by manually inserting an error. Thanks!

**Review by dgn — APPROVED** [review]

**Review by rcernich — APPROVED** [review]

**Pull request merged**